packages/fleather/lib/src/rendering/editor.dart

```diff
@@ -885,7 +885,10 @@ class RenderEditor extends RenderEditableContainerBox
         final caretOffset = child.getOffsetForCaret(localPosition);
         final testPosition = TextPosition(offset: sibling.node.length - 1);
         final testOffset = sibling.getOffsetForCaret(testPosition);
-        final finalOffset = Offset(caretOffset.dx, testOffset.dy);
+        // The addition of 1 point to testOffset.dy is added because somehow in
+        // Flutter 3.22 the text painer for the offset is yielding wrong
+        // text position (1 line above the correct line).
+        final finalOffset = Offset(caretOffset.dx, testOffset.dy + 1);
```

Comment:
Really strange, no explanation for this?

Reply:
Couldn't find any. I even took a quick look at the commits in framework and engine.